bot/cogs/watchchannels/bigbrother.py

```diff
@@ -0,0 +1,118 @@
+
+log = logging.getLogger(__name__)
+
+
+class BigBrother(WatchChannel):
+    """User monitoring to assist with moderation"""
+
+    def __init__(self, bot) -> None:
+        super().__init__(
+            bot,
+            destination=Channels.big_brother_logs,
+            webhook_id=Webhooks.big_brother,
+            api_endpoint='bot/infractions',
+            api_default_params={'active': 'true', 'type': 'watch', 'ordering': '-inserted_at'},
+            logger=log
+        )
+
+    @group(name='bigbrother', aliases=('bb',), invoke_without_command=True)
+    @with_role(Roles.owner, Roles.admin, Roles.moderator)
+    async def bigbrother_group(self, ctx: Context) -> None:
+        """Monitors users by relaying their messages to the BigBrother watch channel"""
+        await ctx.invoke(self.bot.get_command("help"), "bigbrother")
+
+    @bigbrother_group.command(name='watched', aliases=('all', 'list'))
+    @with_role(Roles.owner, Roles.admin, Roles.moderator)
+    async def watched_command(self, ctx: Context, update_cache: bool = True) -> None:
+        """
+        Shows the users that are currently being monitored in BigBrother.
+
+        The optional kwarg `update_cache` can be used to update the user
+        cache using the API before listing the users.
+        """
+        await self.list_watched_users(ctx, update_cache)
+
+    @bigbrother_group.command(name='watch', aliases=('w',))
+    @with_role(Roles.owner, Roles.admin, Roles.moderator)
+    async def watch_command(self, ctx: Context, user: Union[User, proxy_user], *, reason: str) -> None:
+        """
+        Relay messages sent by the given `user` to the `#big-brother-logs` channel.
+
+        A `reason` for adding the user to BigBrother is required and will displayed
+        in the header when relaying messages of this user to the watchchannel.
+        """
+        if user.bot:
+            e = Embed(
+                description=f":x: **I'm sorry {ctx.author}, I'm afraid I can't do that. I only watch humans.**",
+                color=Color.red()
+            )
+            await ctx.send(embed=e)
+            return
+
+        if not await self.fetch_user_cache():
+            log.error("Failed to update user cache; can't watch user {user}")
+            return
+
+        if user.id in self.watched_users:
+            e = Embed(
+                description=":x: **The specified user is already being watched**",
+                color=Color.red()
+            )
+            await ctx.send(embed=e)
+            return
+
+        response = await post_infraction(
+            ctx, user, type='watch', reason=reason, hidden=True
+        )
+        if response is not None:
+            self.watched_users[user.id] = response
+            e = Embed(
+                description=f":white_check_mark: **Messages sent by {user} will now be relayed to BigBrother**",
+                color=Color.green()
+            )
+            await ctx.send(embed=e)
+            return
+
```

Comment:
do we not need an else case?

Reply:
No, the `post_infraction` utility function already catches the Exception, sends an error message to the `Context` and then returns `None`. So, both notifying the user and logging the issue are taken care of there.